modules/programs/atuin.nix

```diff
@@ -0,0 +1,157 @@
+        If enabled, this will bind <literal>ctrl-r</literal> to open the Atuin
+        history.
+      '';
+    };
+
+    enableZshIntegration = mkEnableOption "Zsh integration" // {
+      default = true;
+      description = ''
+        Whether to enable Atuin's Zsh integration.
+        </para><para>
+        If enabled, this will bind <literal>ctrl-r</literal> and the up-arrow
+        key to open the Atuin history.
+      '';
+    };
+
+    config = mkOption {
+      type = with types;
+        let
+          prim = oneOf [ bool int str ];
+          primOrPrimAttrs = either prim (attrsOf prim);
+          entry = either prim (listOf primOrPrimAttrs);
+          entryOrAttrsOf = t: either entry (attrsOf t);
+          entries = entryOrAttrsOf (entryOrAttrsOf entry);
+        in attrsOf entries // { description = "Atuin configuration"; };
+      default = { };
+      example = literalExample ''
+        {
+          # where to store your database, default is your system data directory
+          # mac: ~/Library/Application Support/com.elliehuxtable.atuin/history.db
+          # linux: ~/.local/share/atuin/history.db
+          db_path = "~/.history.db";
+
+          ## where to store your encryption key, default is your system data directory
+          key_path = "~/.key";
+
+          ## where to store your auth session token, default is your system data directory
+          session_path = "~/.key";
+
+          ## date format used, either "us" or "uk"
+          dialect = "uk";
+
+          ## enable or disable automatic sync
+          auto_sync = true;
+
+          ## how often to sync history. note that this is only triggered when a command
+          ## is ran, so sync intervals may well be longer
+          ## set it to 0 to sync after every command
+          sync_frequency = "5m";
+
+          ## address of the sync server
+          sync_address = "https://api.atuin.sh";
+
+          ## which search mode to use
+          ## possible values: prefix, fulltext, fuzzy
+          search_mode = "prefix";
+        }
+      '';
+      description = ''
+        Configuration written to
+        <filename>~/.config/atuin/config.toml</filename>.
+        </para><para>
+        See <link xlink:href="https://github.com/ellie/atuin/blob/main/docs/config.md" /> for the full list
+        of options.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable (mkMerge [
+    # Always add the configured `atuin` package.
+    {
+      home.packages = [ cfg.package ];
+    }
+
+    # If there are user-provided settings, generate the config file.
+    (mkIf (cfg.config != { }) {
+      xdg.configFile."atuin/config.toml" = {
+        source = tomlFormat.generate "atuin-config" cfg.config;
+      };
+    })
+
+    (mkIf cfg.enableBashIntegration (let
+      # This adds hooks which are required for `atuin`'s bash integration.
+      # See https://github.com/ellie/atuin#bash
+      bashPreexecPkg = { stdenv, lib, pkgs }:
+        stdenv.mkDerivation rec {
```

Comment:
Definitively move the package to Nixpkgs, even if it's a single file bash script. Then it will be generally useful and we don't take on the packages maintainership burden in HM.